orchestra/static/orchestra/timing/timecard/partials/timecard.html

```diff
@@ -1,10 +1,20 @@
-<div class="timecard-view">
+<div class="timecard-view" ng-if="!vm.dataLoading">
+  <span class="edit-label">
+    Date range
+  </span>
+  <date-picker date="vm.minDate"></date-picker> – <date-picker date="vm.maxDate" max-date="vm.weekEnd"></date-picker>
   <div class="entries">
     <div class="date-group" ng-repeat="dateEntries in vm.timeEntries.entriesByDate | toArray | orderBy:vm.datetimeFromKey:'true'">
       <h2>
         <span class="date-label">{{ vm.prettyDate(dateEntries.$key) }}</span>
+        <span class="day-name">{{ vm.dayName(dateEntries.$key) }}</span>
         <i ng-click="vm.timeEntries.createEntry(vm.datetimeFromKey(dateEntries))" class="fa fa-plus btn-add-entry"></i>
-        <span class="date-duration">{{ vm.timeEntries.timeWorkedForDate(dateEntries.$key).humanizeUnits() }}</span>
+        <span class="date-duration" ng-show="vm.timeEntries.invalidEntriesForDate(dateEntries.$key).length">
+          <em>Invalid Entries</em>
```

Comment:
i guess it's more incomplete than invalid---the user can fix the invalidity into completion